### ISSUE TITLE
Cache always delivering old versions

### ISSUE THREAD

**Issue by FreeTrade** [memberapp/memberapp.github.io#261]
Cache doesn't seem to update to newer versions. Always need to shift+reload to get new version, but on next load, cache reverts back to old version. Guess cache should update with latest version of page, and when version name changes, cache should reload all resources.
@2qx

**Comment by 2qx** [comment]
When a new version is released, the browser should:

 - check and see that a new version of the service worker is available 
 - Download the new sw.js and make is available on next page load. 
 - The new service worker installs into a new cache.  

I think service worker might be commented out, and before that it may have been the issue with the manifest.  

I was getting updates until around Jan 5th

**Comment by FreeTrade** [comment]
When the PWA is installed, this issue seems to re-occur. I've several installs on v3.6.1 to won't update to v3.7

**Comment by 2qx** [comment]
When I load the current v3.7 I get 

```
The FetchEvent for "https://memberapp.github.io/index.html" resulted in a network error response: the promise was rejected.
sw.js:77 Uncaught (in promise) TypeError: Failed to execute 'fetch' on 'WorkerGlobalScope': 'only-if-cached' can be set only with 'same-origin' mode
```
Which would mean it's not able to cache the new index.html. 

Let me rebase see if I can get my v3.5 installs to upgrade.

**Comment by 2qx** [comment]
I set my master branch to your v3.7, and the v3.5 apps I had installed from my gitpages updated fine.  Regardless of the issue above, when the service worker installs it should delete all old caches.

The only caveat I can think of is that on initial load, the app should check and see that there is a new sw.js and download it in the background, and only on the next reload will the new app be loaded.  

I assume you tried a reload though.
